test/e2e/misc_test.go

```diff
@@ -88,6 +87,48 @@ func (suite *MiscTestSuite) TestValidateBuildImage() {
 	}
 }
 
+// This is a test of the findRoute function, not a product test
+func (suite *MiscTestSuite) TestFindRoute() {
+	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval}
+
+	jaegerInstanceName := "simplest"
+	jaegerInstance := getSimplestJaeger(jaegerInstanceName, namespace)
+	err := fw.Client.Create(context.Background(), jaegerInstance, cleanupOptions)
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(jaegerInstance)
+
+	// Create a second namespace and deploy another instance named "simplest"
+	secondContext, err := prepare(t)
+	if err != nil {
+		if secondContext != nil {
+			secondContext.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare with: "+err.Error())
+	}
+	defer secondContext.Cleanup()
+
+	secondNamespace, err := secondContext.GetNamespace()
+	secondJaegerInstance := getSimplestJaeger(jaegerInstanceName, secondNamespace)
+	err = fw.Client.Create(context.Background(), secondJaegerInstance, cleanupOptions)
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(secondJaegerInstance)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, secondNamespace, jaegerInstanceName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for deployment")
+
+	// Make sure findRoute returns the correct routes for each namespace
+	route := findRoute(t, fw, jaegerInstanceName, namespace)
+	require.Equal(t, namespace, route.Namespace)
+	require.True(t, strings.Contains(route.Spec.Host, namespace))
```

Comment:
Shouldn't the host be the jaeger instance name? (or part of it, if the namespace is too long)?

Reply:
@objectiser Do you mean route.Spec.Host?  That's actually what I would think of as the whole route, minus "https://".  i.e.  simplest-osdk-e2e-70751771-8736-4b0c-98ea-909f7cd22a86.apps.kearls-c435d.devcluster.openshift.com

Reply:
Sorry, correct.